parsl/executors/high_throughput/process_worker_pool.py

```diff
@@ -603,6 +603,10 @@ def update_resource_spec_env_vars(mpi_launcher: str, resource_spec: Dict, node_i
 
 
 def _init_mpi_env(mpi_launcher: str, resource_spec: Dict):
+    for varname in resource_spec:
```

Comment:
I suspect that env vars could persist over to the following tasks because we do not explicitly clear them. This is unrelated to this PR.